pandas/io/formats/format.py

```diff
@@ -1514,18 +1514,25 @@ def __init__(
 
     def _format_strings(self) -> List[str]:
         """ we by definition have DO NOT have a TZ """
-        values = self.values
-
-        if not isinstance(values, DatetimeIndex):
-            values = DatetimeIndex(values)
+        values = np.asarray(self.values)
+        flat_values = values.ravel() if values.ndim > 1 else values
+        flat_values = DatetimeArray(flat_values)
```

Comment:
DatetimeArray can handle 2D values, so some of the ravel/ndim checks might not be needed

Reply:
Yes I noticed it's able to handle 2D values, but the extension array that I was fixing this for can have higher dimensions, so it would still need to check above that plus when `self.formatter` is set. It seems cleaner to flatten at the beginning and work with flattened values. And I think calling `ravel()` a second time would be a no-op?

Reply:
OK no problem